File: 0957-minimum-add-to-make-parentheses-valid/0957-minimum-add-to-make-parentheses-valid.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        int count = 0;
        stack<int> st;
        for(char ch : s){
            if(ch == ')'){
                if(st.empty())
                    count++;
                else if(st.top() == '(')
                    st.pop();
                else 
                    st.push(ch);
            } else if(ch == '('){
                st.push(ch);
            }
        }
        return count+st.size();
    }
};

File: 0957-minimum-add-to-make-parentheses-valid/0957-minimum-add-to-make-parentheses-valid.py
class Solution(object):
    def minAddToMakeValid(self, s):
        """
        :type s: str
        :rtype: int
        """
        open_brackets = 0
        close = 0
        for char in s:
            if char == '(':
                open_brackets += 1
            else :
                if open_brackets > 0:
                    open_brackets -= 1
                else:
                    close += 1
        return open_brackets + close

        